IS0401Test.py

```diff
@@ -581,15 +577,23 @@ def test_17(self):
 
         return test.PASS()
 
-    def do_receiver_put(self, receiver_id, data):
+    def do_receiver_put(self, test, receiver_id, data):

```

Comment:
I have been wondering about doing this refactoring in a few places, e.g. IS0501Test ``check_`` methods. What do you think?

Reply:
In terms of using the exception to trigger a failure? Yes, Simon suggested adding it for IS-08 work but I'm sure it's relevant in lots of other cases if the only thing we're doing is cascading a failure back to be immediately shown as a negative result.

Reply:
Yes, having check methods that have a Test argument and which are either successful or raise an exception to get some kind of failure out directly. The cases that really motivated me were the string checks like e.g. https://github.com/AMWA-TV/nmos-testing/blob/3bdd1e5694946192e5c0126ed3dcd17c2162248b/IS0501Test.py#L238